setup1-contributor/src/commands/contribute.rs

```diff
@@ -222,6 +226,34 @@ impl Contribute {
             delay_for(Duration::seconds(DELAY_WAIT_FOR_PIPELINE_SECS).to_std()?).await;
         }
 
+        let cloned = self.clone();
+        //TODO: this is broken, probably because there is some
```

Comment:
I still need to confirm why this is broken, if it is indeed blocking code in the async path, then I should probably report a bug. However for now it's probably not worth spending time to discover/fix precisely which code is blocking.

Reply:
left a note in there about why I'm using a thread, won't investigate further for now